Source/Core/DolphinQt/ToolBar.cpp

```diff
@@ -14,14 +14,28 @@
 #include "DolphinQt/Settings.h"
 #include "DolphinQt/ToolBar.h"
 
-static QSize ICON_SIZE(32, 32);
+static const QSize ICON_SIZE(32, 32);
+static const QSize ICON_SIZE_SMALL(16, 16);
 
 ToolBar::ToolBar(QWidget* parent) : QToolBar(parent)
 {
-  setToolButtonStyle(Qt::ToolButtonTextUnderIcon);
+  auto update_compact_view_lambda = [this] {
+    if (Settings::Instance().IsCompactViewEnabled())
+    {
+      setIconSize(ICON_SIZE_SMALL);
+      setToolButtonStyle(Qt::ToolButtonIconOnly);
+    }
+    else
+    {
+      setIconSize(ICON_SIZE);
+      setToolButtonStyle(Qt::ToolButtonTextUnderIcon);
+    }
+  };
+  connect(&Settings::Instance(), &Settings::CompactViewChanged, this, update_compact_view_lambda);
+  update_compact_view_lambda();
```

Comment:
I think it might be better to turn this lambda into it's own method:

1. It's not only used inline but called separately here
2. Tidier
3. Nicer ``connect`` syntax